modules/openapi-generator/src/main/resources/python-legacy/api_client.mustache

```diff
@@ -545,10 +545,12 @@ class ApiClient(object):
         else:
             return ', '.join(accepts)
 
-    def select_header_content_type(self, content_types):
+    def select_header_content_type(self, content_types, method=None, body=None):
```

Comment:
Why have you added the parameters method and body here when they re unused in the method?

Reply:
Thanks for this question. To be backward compatible I don't provide an extra code here but these parameters help me to do it later (patch or subclass the ApiClient).

But maybe we can "break" it a little and add support for JSON Patch:

```
if method == 'PATCH' and isinstance(body, list) and 'application/json-patch+json' in content_types:
   return 'application/json-patch+json'
```

What do you think?

Reply:
It doesn't make sense to add parameters that are not used by our client because other users want to override them.
Why not pass in _content_type = 'application/json-patch+json' for your mentioned case?

Reply:
To be more precisely - I use openapi-generator to build my client library. I'd like to prepare my own select_header_content_type which has a custom detection and uses the body and method (for instance I can check if it's json or not, it contains list or not etc.). Even if this original method doesn't use them, my method can.

My end-user, who uses my library, can pass _content_type to manually force the content-type, as you said and it's also important.

In this PR I'd like to provide these two options - forcing content-type and possibility to provide custom detection/selection.

Reply:
I've decided to add support for 'application/json-patch+json' where these parameters are used.

Reply:
So I don't like adding parameters that our code base that we don't use but I understand why you need them.
Can you add a test that patches select_header_content_type and verifies that it is called with all of your needed inputs?
If you don't add this then someone could edit this feature in the future and remove the unused parameters like body and break your feature.

Reply:
Sure, I've just added tests.